var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -652,6 +667,25 @@ def add_buildopt_symlinks(self, prefix):
                     prefix, remainder = lib.split(".", 1)
                     symlink(lib, "%s-mt.%s" % (prefix, remainder))
 
+    def boostrap_windows(self):
+        """Run the Windows-specific bootstrap.bat. The only bootstrapping command
+        line option that is accepted by the bootstrap.bat script is the compiler
+        information: either the vc version (e.g. MSVC 14.3.x would be vc143)
+        or gcc or clang.
+        """
+        bootstrap_options = list()
+        if self.spec.satisfies("%msvc"):
+            # MSVC is the default compiler and is automatically detected by
+            # the bootstrapping script.
+            pass
+        elif self.spec.satisfies("%gcc"):
+            bootstrap_options.append("gcc")
+        elif self.spec.satisfies("%clang"):
+            bootstrap_options.append("clang")
+
+        bootstrap = Executable("cmd.exe")
+        bootstrap("/c", ".\\bootstrap.bat", *bootstrap_options)
```

Comment:
> Things that determine_bootstrap_options was handling that I don't see handled here:
> 
> +python
> +icu
>
> Are those not supported on Windows? We may also want to add conflicts for those.

* `icu` requires autotools, so I will add a conflict for windows
* `python` works, I will put in a couple of changes to get it working on windows


 
>Overall it may be slightly better to make the flow consistent (right now the Windows path calls the do-this function, and the Linux path calls the set-everything-up-for-me-and-then-I'll-do-it function, I recommend they both follow the same high-level approach.

Yes I agree, but I was trying to get Windows working without making any changes that might affect Linux users since Boost is such a fundamental package. The difference in bootstrapping between Linux and Windows is bizarre to me.

Reply:
@scheibelp do you want the `icu` support here? I have a patch that ports ICU to Windows that's been sitting in my QT port branch.

Reply:
No, I tried to add this flag and the build fails with it in the command line.